octree_web_viewer/src/state.rs

```diff
@@ -0,0 +1,95 @@
+use crate::backend_error::PointsViewerError;
+use point_viewer::octree;
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, RwLock};
+
+/// path information for the octrees
+#[derive(Clone)]
+pub struct OctreeKeyParams {
+    /// Location prefix, including
+    prefix: PathBuf,
+    /// Tree ID
+    suffix: PathBuf,
+}
+
+impl OctreeKeyParams {
+    pub fn get_octree_address(
+        &self,
+        octree_key: impl AsRef<Path>,
+    ) -> Result<PathBuf, PointsViewerError> {
```

Comment:
Do we really need a `Result` with an error here? I guess returning `PathBuf` would be enough, as the method internally doesn't give any error...

Reply:
done!